tests/x11regressions/empathy/empathy_irc.pm

```diff
@@ -48,16 +48,13 @@ sub run() {
     };
 
     # join openqa channel
-    if (sle_version_at_least('12-SP2')) {
+    send_key "ctrl-j";
+    if (!check_screen 'empathy-join-room', 4) {
```

Comment:
To prevent these kind of race conditions so far I did not see an example which could be prevented by making the needle more restrictive, maybe you just need multiple matching areas? Sorry if I don't understand what the problem is. Maybe you can help me with pointing me to a screenshot of a failing job and the corresponding screenshot of your local runs where you fixed it?

Otherwise, how about

```perl
# join openqa channel
wait_screen_change { send_key 'ctrl-j' };
if (!check_screen 'empathy-join-room', 0) {
    record_soft_failure 'bsc#999832: keyboard shortcut of empathy not working on SLED12SP2';
    assert_and_click 'empathy-menu';
    send_key_until_needlematch 'empathy-menu-rooms', 'down';
    assert_and_click 'empathy-menu-joinrooms';
}
```